homeassistant/components/device_tracker/mercedesme.py

```diff
@@ -0,0 +1,62 @@
+"""
+Support for Mercedes cars with Mercedes ME.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/mercedesme/
+"""
+import logging
+from homeassistant.helpers.event import track_utc_time_change
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_MME = 'mercedesme'
+
+
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up the Mercedes ME tracker."""
+    if discovery_info is None:
+        return False
+
+    controller = hass.data[DATA_MME]['controller']
+
+    if not controller.cars:
+        return False
+
+    MercedesMEDeviceTracker(hass, config, see, controller)
+
+    return True
+
+
+class MercedesMEDeviceTracker(object):
+    """A class representing a Mercedes ME device tracker."""
+
+    def __init__(self, hass, config, see, controller):
+        """Initialize the Mercedes ME device tracker."""
+        self.hass = hass
+        self.see = see
+        self.controller = controller
+        self.update_info()
+
+        track_utc_time_change(

```

Comment:
Instead of registering the update yourself, inherit from `device_tracker.DeviceScanner`, and implement `get_scanner` instead of `setup_scanner`.

Example: https://github.com/home-assistant/home-assistant/blob/46e030662da4ab7c653a23e1f34e527c9a0e7a8f/homeassistant/components/device_tracker/bbox.py

Reply:
Thank you @armills for the review. I've taken your points and changed the code. Only this comment takes longer. All current implementations that are using the `device_tracker.DeviceScanner` class have not implemented GPS coordinates. Could it be that the DeviceScanner is for local network devices and the DeviceTracker for devices that are not directly connected to the local network? All other car components using `setup_scanner`. If it's ok for you, I'll check, learn and implement this in the next cycle. thx

Reply:
Nice work with the follow up!

I'm OK holding off on the larger refactor, but we should at least switch to `track_time_interval` with the configured time interval. There's an example of an invocation below. It should be a pretty quick change.

https://github.com/home-assistant/home-assistant/blob/93c8372c30200071d54a9e1710805d2bc91688a3/homeassistant/components/raincloud.py#L123